src/rust/engine/fs/store/src/snapshot_ops.rs

```diff
@@ -229,7 +228,7 @@ pub trait SnapshotOps: Clone + Send + Sync + 'static {
   ) -> Result<DirectoryDigest, Self::Error> {
     let input_tree = self.load_digest_trie(directory_digest.clone()).await?;
     let path_stats = input_tree
-      .expand_globs(params.globs, SymlinkBehavior::Oblivious, None)
+      .expand_globs(params.globs, SymlinkBehavior::Aware, None)
       .await
       .map_err(|err| format!("Error matching globs against {directory_digest:?}: {}", err))?;
```

Comment:
Hm. So... this is an interesting case. This is implementing glob matching, and when users match `PathGlobs` against their filesystem, they're expecting them to be symlink oblivious... I think that this is probably the same case?

But the `DigestTrie` should be constructed from the matched `Stat` objects, rather than their symbolic names, I think? Hm.

Reply:
In the end, I think the end result here is that they will get symlink-oblivious data when using symlink-oblivious avenues (`.files` on a snapshot) but the resulting Digest should still be the most correct subset (i.e. symlink aware)

Reply:
> but the resulting Digest should still be the most correct subset (i.e. symlink aware)

I don't think so... if I try to match `a/file.txt` and `a` is a symlink, then it won't ever match in a symlink-aware situation, because the symlink won't be traversed.

Reply:
Oh I see. Yeah this one is a head-scratcher.

Reply:
Tangent, but: I think that's even more reason not to promise glob expansion for `Process.output_files`... if we are symlink aware, I don't think that we can actually do it. So `output_paths` probably. cc @tdyas

Reply:
So would you advocate for changing `DigestSubset` to not use globs? I can start on that

Reply:
No: I'm suggesting that `DigestSubset` should be symlink oblivious, in order to act the same way that `PathGlobs` do elsewhere in the Python API.

Reply:
Oh yeah hmm I guess the only place we'd see funny-business would be someone subsetting a Process' output.

Seems stomachable

Reply:
Yea... funny business is definitely possible. I'm not exactly sure how to handle this, but this seems like our best option for now...?

One option in the medium term would maybe be to expose obliviousness on `PathGlobs` itself?